plugin/trino-elasticsearch/src/test/java/io/trino/plugin/elasticsearch/BaseAggregationPushDownTest.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.elasticsearch;
+
+import com.google.common.collect.ImmutableMap;
+import io.trino.sql.planner.plan.FilterNode;
+import org.testng.annotations.Test;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public abstract class BaseAggregationPushDownTest
+        extends BaseElasticsearchConnectorTest
+{
+    public BaseAggregationPushDownTest(String image)
+    {
+        super(image);
+    }
+
+    @Test
+    public void testAggregationPushdown()
+    {
+        // SELECT DISTINCT
+        assertThat(query("SELECT DISTINCT regionkey FROM nation")).isFullyPushedDown();
+
+        // count()
+        assertThat(query("SELECT count(*) FROM nation")).isFullyPushedDown();
+        assertThat(query("SELECT count(nationkey) FROM nation")).isFullyPushedDown();
+        assertThat(query("SELECT count(1) FROM nation")).isFullyPushedDown();
+        assertThat(query("SELECT count() FROM nation")).isFullyPushedDown();
+
+        // GROUP BY
+        assertThat(query("SELECT regionkey, min(nationkey) FROM nation GROUP BY regionkey")).isFullyPushedDown();
+        assertThat(query("SELECT regionkey, max(nationkey) FROM nation GROUP BY regionkey")).isFullyPushedDown();
+        assertThat(query("SELECT regionkey, sum(nationkey) FROM nation GROUP BY regionkey")).isFullyPushedDown();
+        assertThat(query("SELECT regionkey, avg(nationkey) FROM nation GROUP BY regionkey")).isFullyPushedDown();
```

Comment:
Please check `testKeywordAggregationWithEmptyBucketPushdown` which includes null column, and multiple group by fields. 
For tcpds, there is no suitable table to do this test.

Reply:
Right, I was thinking we could extend the index you created for that test with some additional numeric columns and populate those with both null and non-null values.  And use that table instead of `nation`.  The idea is to ensure that the semantics of pushed down operations matches those of Trino for edge cases involving nulls.

I was just playing with this and noticed that when running a `sum()` over values that are all null, ES returns 0.  When the operation is not pushed down and is performed by Trino, the result is `null`.  I am actually not sure how to deal with this mismatch, as there doesn't seem to be a way to distinguish between summing say all 0s and all nulls.  Incidentally, `avg()` has semantics that match Trino's (i.e. it returns `null` when run over a set of values that are all `null`).

Reply:
One hacky way to deal with the limitation of `sum()` for the all nulls scenario could be to compute `avg()` over the same column, as well, and if the average comes back as `null`, then we would know that the sum would have to be `null`, as well.  It's ugly and adds overhead, perhaps there are better workarounds.

Reply:
Oh, that's a tough issue. There seems to be no good solution. Es officials seem to want to keep sum's result 0